src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/ResourceReferenceResolver.cs

```diff
@@ -74,6 +74,10 @@ public async Task ResolveReferencesAsync(Resource resource, IDictionary<string,
                         {
```

Comment:
This check needs to be changed to be where results equal 0 instead of not equal to 1, otherwise the new else if won't be reached.